features/discount/handler/handler.go

```diff
@@ -0,0 +1,135 @@
+package handler
+
+import (
+	"net/http"
+	discountModel "star-pos/features/discount/model"
+	"star-pos/features/discount/service"
+	"star-pos/utils/response"
+	"strings"
+
+	"github.com/labstack/echo/v4"
+)
+
+func CreateDiscounts(c echo.Context) error {
+	newDiscount := discountModel.Discount{}
+	errBind := c.Bind(&newDiscount)
+	if errBind != nil {
+		return c.JSON(http.StatusInternalServerError, response.WebJSONResponse("error bind data: "+errBind.Error(), nil))
+	}
+
+	idCreated, err := service.Add(newDiscount)
+	if err != nil {
+		if strings.Contains(err.Error(), "user id") {
+			return c.JSON(http.StatusBadRequest, response.WebJSONResponse("error create Discount: "+err.Error(), nil))
+		} else if strings.Contains(err.Error(), "name") {
+			return c.JSON(http.StatusBadRequest, response.WebJSONResponse("error create Discount: "+err.Error(), nil))
+		} else {
+			return c.JSON(http.StatusInternalServerError, response.WebJSONResponse("error create Discount: "+err.Error(), nil))
+		}
+	}
+
+	result := response.ResponseCreate{
+		ID: idCreated,
+	}
+
+	return c.JSON(http.StatusCreated, result)
+}
+
+func GetAllDiscounts(c echo.Context) error {
+	result, err := service.GetAllDiscounts()
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, response.WebJSONResponse("error get all Discounts: "+err.Error(), nil))
+	}
+
+	var responseDiscounts []DiscountsResponse
+
+	for _, data := range result {
+		responseDiscounts = append(responseDiscounts, DiscountsResponse{
+			ID:           data.ID,
+			UserID:       data.UserID,
+			OutletID:     data.OutletID,
+			ProductID:    data.ProductID,
+			DiscountName: data.DiscountName,
+			Amount:       data.Amount,
+			DiscountType: data.DiscountType,
+			DiscountMax:  data.DiscountMax,
+			CreatedAt:    data.CreatedAt,
+			UpdatedAt:    data.UpdatedAt,
+			User:         data.User,
```

Comment:
I think it's no needed. Because it's use for foreign key of userID. As well as outlet and product. So it doesn't to be shown